src/v/kafka/server/handlers/list_offsets.cc

```diff
@@ -132,7 +132,7 @@ static ss::future<list_offset_partition_response> list_offsets_partition(
     auto res = co_await kafka_partition->timequery(storage::timequery_config{
       kafka_partition->start_offset(),
       timestamp,
-      offset,
+      model::prev_offset(offset),
```

Comment:
Yikes, nice catch!

Reply:
I'm wondering what the implications of this are. Presumably if the `offset` doesn't exist, the timequery will just return early, and if it does exist, it won't affect the result of a timequery. Is that right?

Maybe a good idea to add any implications (or lackthereof) into the commit message

Reply:
> and if it does exist, it won't affect the result of a timequery. Is that right?

You mean prior to this commit? It would have returned a potentially non-committed offset even with acks=all. Normally, redpanda doesn't expose such offsets unless write caching is in use. Fetch would have failed with out of range too, causing the consumer to reset.